torch/library_stable.h

```diff
@@ -0,0 +1,92 @@
+// this file can only have stable stuff! Akin to shim.h
+// but unlike shim.h, this file can contain header-only C++
+// code for better UX.
+
+#include <torch/csrc/inductor/aoti_torch/c/shim.h>
+
+class TORCH_API StableLibrary final {
+  private:
+    TorchLibraryHandle lib_;
+  public:
+    // a kind
+    enum Kind {
+      // DEF, // from TORCH_LIBRARY (no qualifier)
+      IMPL,
+      // FRAGMENT,
+    };
+
+    // constructor
+    /// \private
+    ///
+    /// Use STABLE_TORCH_LIBRARY or STABLE_TORCH_LIBRARY_IMPL() instead of using these
+    /// constructors directly
+    StableLibrary(
+        Kind kind,
+        const char* ns,
+        const char* k,
+        const char* file,
+        uint32_t line) {
+      if (kind==IMPL) {
+        aoti_torch_library_init_for_impl(ns, k, file, line, &lib_);
+      } else {
+        std::cout << "ERROR: StableLibrary constructor not yet implemented for kind=" << kind << std::endl;
+      }
+    }
+
+    StableLibrary(const StableLibrary&) = delete;
+    StableLibrary& operator=(const StableLibrary&) = delete;
+    StableLibrary(StableLibrary&&) = default;
+    StableLibrary& operator=(StableLibrary&&) = default;
+    ~StableLibrary() = default;
```

Comment:
As discussed offline, we want this to call into something akin to aoti_torch_library_delete